src/help/help_impl.cpp

```diff
@@ -623,6 +625,65 @@ std::vector<topic> generate_faction_topics(const config & era, const bool sort_g
 	return topics;
 }
 
+std::vector<topic> generate_trait_topics(const bool sort_generated)
+{
+	std::vector<topic> topics;
+	std::map<t_string, const config> trait_list;
+
+	for (const config & trait : unit_types.traits()) {
+		const std::string trait_id = trait["id"];
+		trait_list.insert(std::make_pair(trait_id, trait));
+	}
+
+
+	for (const unit_type_data::unit_type_map::value_type &i : unit_types.types())
+	{
+		const unit_type &type = i.second;
+		if (description_type(type) == FULL_DESCRIPTION) {
```

Comment:
Is this the check for whether the unit has been seen or the pref is set to show all units?